### PR TITLE
fixed how Errors are handled

### PR DESCRIPTION
**Pull request by thorix**
Errors where double printing part of the message. We found that parseError was being called two times creating this output: `2019-05-17 10:54:03.870426	[ FATAL ]	Vault Error: Vault Error: Get...`

### CODE REVIEW

File: pkg/vault/error.go [hunk 1]
```diff
@@ -11,22 +11,39 @@ import (
...
-type Error struct {
+// CustomVaultError is the custom error type for this package
+type CustomVaultError struct {
 	MessageParts  []string
 	OriginalError error
 }
 
+// Error is custom error interface with one method that will block other functions
+// from using this Error. This is not interchangeable with the standard error.
+type Error interface {
```
Comment (lwahlmeier) [inline pkg/vault/error.go]: Personally I would change this error name to, could possibly get away with keeping `CustomVaultError` private `customVaultError` and then only using the interface everywhere

File: pkg/vault/error.go [hunk 2]
```diff
@@ -11,22 +11,39 @@ import (
...
-	var verr Error
-	verr.OriginalError = err
+	// Provent parseError from calling parseError again
+	if serr, ok := err.(Error); ok {
+		return serr
+	}
+
+	// var verr CustomVaultError
+	verr := &CustomVaultError{OriginalError: err}
+	// verr.OriginalError = err
```
Comment (lwahlmeier) [inline pkg/vault/error.go]: can probably remove these comments

File: pkg/vault/status.go [hunk 3]
```diff
@@ -5,11 +5,11 @@ import (
 	"time"
 )
 
-func (v *Vault) isVaultHealthy() (bool, error) {
+func (v *Vault) isVaultHealthy() (bool, Error) {
 
 	result, err := v.client.Sys().Health()
 	if err != nil {
-		return false, v.parseError(err)
+		return false, v.parseError(err).(Error)
```
Comment (lwahlmeier) [inline pkg/vault/status.go]: Do you really have to cast this?  I would not think so.

File: pkg/vault/token.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ func (v *Vault) GetCurrentTokenTTL() (time.Duration, error) {
 	// Get the token info from Vault
 	secret, err := v.client.Auth().Token().LookupSelf()
 	if err != nil {
-		return 0, err
+		return 0, v.parseError(err)
```
Comment (lwahlmeier) [inline pkg/vault/token.go]: This function should be returning Error not error

File: pkg/vault/error.go (hunk 2 above)
Comment (thorix) [reply pkg/vault/error.go]: fixed

File: pkg/vault/status.go (hunk 3 above)
Comment (thorix) [reply pkg/vault/status.go]: fixed

**Review by bartlettc22 — APPROVED** [review]

**Pull request merged**